src/wwwroot/controllers/MainCtrl.js

```diff
@@ -63,7 +65,22 @@
         isFreeTrialAlmostEnded : false,
         isFreeTrialEndToday : false
       }
-      
+      if(!auth.getFreeTrialNotificationFromStorage() && !$rootScope.modalOpened) {
+        $rootScope.modalOpened = true;
+        ModalService.showModal({
+          templateUrl: 'partials/modals/general-template.html',
+          controller: 'GeneralTemplateCtrl',
+          controllerAs: 'vm',
+          inputs: {
+            title: "free_trial_ended_popup_title",
+            mainText: "free_trial_ended_popup_subtitle",
+            buttonText: "free_trial_ended_popup_button_text",
+            action: freeTrialNotificated
+          }
+        }).then(function (modal) {
+          modal.close.then(toggleModalStatus);
```

Comment:
Is it not possible to define `modalOpened` as local variable in place of a `$rootScope`attribute.